controllers/factory/k8stools/expansion.go

```diff
@@ -139,24 +140,46 @@ func getPVCFromSTS(pvcName string, sts *appsv1.StatefulSet) *corev1.PersistentVo
 	return pvc
 }
 
-func growSTSPVC(ctx context.Context, rclient client.Client, sts *appsv1.StatefulSet, pvcName string) error {
-	pvc := getPVCFromSTS(pvcName, sts)
-	if pvc == nil {
-		// fast path
-		var names []string
-		for i := range sts.Spec.VolumeClaimTemplates {
-			names = append(names, sts.Spec.VolumeClaimTemplates[i].Name)
-		}
-		log.Error(fmt.Errorf("cannot find PVC by name: %s for sts: %s, looks like bug, exist names for sts: %s", pvcName, sts.Name, strings.Join(names, ",")), "cannot find pvc to grow")
-		return nil
+func growSTSPVC(ctx context.Context, rclient client.Client, sts *appsv1.StatefulSet) error {
+	targetPVCs := sts.Spec.VolumeClaimTemplates
+	// list current pvcs
+	var pvcs corev1.PersistentVolumeClaimList
+	opts := &client.ListOptions{
+		Namespace:     sts.Namespace,
+		LabelSelector: labels.SelectorFromSet(sts.Spec.Selector.MatchLabels),
 	}
-	// check storage class
-	isExpandable, err := isStorageClassExpandable(ctx, rclient, pvc)
-	if err != nil {
+	if err := rclient.List(ctx, &pvcs, opts); err != nil {
 		return err
 	}
-
-	return growPVCs(ctx, rclient, pvc.Spec.Resources.Requests.Storage(), sts.Namespace, sts.Spec.Selector.MatchLabels, isExpandable)
+	if len(pvcs.Items) == 0 {
+		log.Info("PVCs select call returned 0 pvcs, it could be a bug, want match %v in namespace %s", sts.Spec.Selector.MatchLabels, sts.Namespace)
+		return nil
+	}
+	for _, pvc := range pvcs.Items {
+		var isExist bool
+		for _, tpvc := range targetPVCs {
+			if strings.HasPrefix(pvc.Name, fmt.Sprintf("%s-%s", tpvc.Name, sts.Name)) {
+				isExist = true
+				// check if storage class is expandable
+				isExpandable, err := isStorageClassExpandable(ctx, rclient, &pvc)
+				if err != nil {
+					logger.Errorf("failed to check storageClass expandability for pvc %s: %v", pvc.Name, err)
```

Comment:
I think, it's better to return errors back to the caller. Otherwise it performs partial update and actual resize may never happen.

Reply:
Right, fixed!
I was wondering when should we interrupt multiple update process by returning once we get an error, or just log it and give the rest obejcts a chance to be updated. 
In this case, maybe it's better to return it, since it's already a partial update, and by returning it, the error message would also be persisted to vmcluster.status.reason)